utils/warnings.ml

```diff
@@ -624,6 +626,9 @@ let message = function
         "This type declaration is defining a new '()' constructor\n\
          which shadows the existing one.\n\
          Hint: Did you mean 'type %s = unit'?" name
+  | Generic_comparison ->
+      "this comparison will use the generic comparison function.\n\
+        It may have unexpected semantics in some cases."
```

Comment:
Rather than the rather vague "It may have unexpected semantics in some cases" (which might be true of anything, depending on your expectations) I think it would be better to say what the unexpected semantics are.  Perhaps "which operates on the low-level representations of values, ignoring types".